whisper.cpp

```diff
@@ -380,16 +380,18 @@ struct whisper_vocab {
     std::map<token, id> token_to_id;
     std::map<id, token> id_to_token;
 
-    id token_eot  = 50256;
-    id token_sot  = 50257;
-    id token_prev = 50360;
-    id token_solm = 50361; // ??
-    id token_not  = 50362; // no timestamps
-    id token_beg  = 50363;
-
-    // available tasks
-    static const id token_translate  = 50358;
-    static const id token_transcribe = 50359;
+    // reference: https://github.com/openai/whisper/blob/248b6cb124225dd263bb9bd32d060b6517e067f8/whisper/tokenizer.py#L334-L349
+    id token_eot        = 50256;
+    id token_sot        = 50257;
+    // task tokens (used only for multilingual models)
+    id token_translate  = 50357;
+    id token_transcribe = 50358;
+    // other special tokens
+    id token_tdrz       = 50359; // [TDRZ] used by tinydiarize models to indicate speaker turn
+    id token_prev       = 50360;
+    id token_solm       = 50361; // start of lm ?
```

Comment:
Hi @ggerganov - so actually id 50361 corresponds to the `nospeech` token.

https://github.com/openai/whisper/blob/248b6cb124225dd263bb9bd32d060b6517e067f8/whisper/tokenizer.py#L334-L349

Both variables `token_solm` and `token_tdrz` should have the same value. This is an unused token ID in the released whisper models https://github.com/openai/whisper/discussions/414.

FYI - there was actually a bug in whisper.cpp with some incorrect token IDs, which I fixed in this commit - https://github.com/akashmjn/whisper.cpp/commit/c8e1ed6f8281e83abc8cbd54483c24c767bcde66

Reply:
Got it - should be fixed now